### PR TITLE
Edit, List and Delete feature added to DietGoalList

### LINKED ISSUE #9: List all diet goals
As a forgetful user, I can see all my diet goals so that I can remind myself of all the goals I have set.

### PR DESCRIPTION
**Pull request by yicheng-toh**
Changes are made to DietGoal and DietGoalList class to support the new features.

Closes #9 , #10 , #11

### CODE REVIEW

File: src/test/java/athleticli/ui/dietgoal/DietGoalTest.java [hunk 1]
```diff
@@ -15,12 +15,27 @@ void getNutrients_initialiseCommonArgs_expectArgs() {
...
+    @Test
+    void setTargetValue_initialiseCommonArgs_expectArgs() {
+        DietGoal proteinGoal = new DietGoal("protein", 10000);
+        proteinGoal.setTargetValue(10);
+        assertEquals(10, proteinGoal.getTargetValue());
+
+    }
+
     @Test
     void getCurrentValue_initialiseCommonArgs_expectZero() {
```
Comment (skylee03) [inline src/test/java/athleticli/ui/dietgoal/DietGoalTest.java:40]: According to the [coding standard](https://se-education.org/guides/conventions/java/basic.html#comments), we are recommended to use American spelling.
```suggestion
    void getCurrentValue_initializeCommonArgs_expectZero() {
```

**Review by skylee03 — CHANGES_REQUESTED** [review]
LGTM! However, according to the [coding standard](https://se-education.org/guides/conventions/java/basic.html#comments), we are recommended to use American spelling. Thus, I think we'd better change "initialise" to "initialize" in the test methods. I have marked the parts that need to be modified, but some of them involve previously merged code and cannot be marked.

**Review by skylee03 — COMMENTED** [review]
LGTM! However, according to the [coding standard](https://se-education.org/guides/conventions/java/basic.html), the getter/setter of `isGoalAchieved` shoule be called `isGoalAchieved()` and `setGoalAchieved()`. This is also the default naming method of IntelliJ IDEA.

File: src/main/java/athleticli/dietgoal/DietGoal.java [hunk 2]
```diff
@@ -25,10 +25,23 @@ public String getNutrients() {
...
         return targetValue;
     }
 
+    public void setTargetValue(int targetValue) {
+        this.targetValue = targetValue;
+        if (!isGoalAchieved && currentValue >= targetValue) {
+            setIsGoalAchieved(true);
+        } else if (isGoalAchieved && currentValue < targetValue) {
+            setIsGoalAchieved(false);
+        }
```
Comment (nihalzp) [inline src/main/java/athleticli/dietgoal/DietGoal.java:42]: Should we simplify the logic and always set depending on the currentValue?
```suggestion
        if (currentValue >= targetValue) {
            setIsGoalAchieved(true);
        } else {
            setIsGoalAchieved(false);
        }
```

File: src/main/java/athleticli/dietgoal/DietGoalList.java [hunk 3]
```diff
@@ -13,9 +13,25 @@ public void addGoal(DietGoal dietGoal) {
...
         return dietGoals.size();
     }
+
+    public String toString() {
+        StringBuilder result = new StringBuilder();
+        for (int i = 0; i < dietGoals.size(); i++) {
+            result.append(i + 1).append(". ").append(dietGoals.get(i).toString());
+        }
+        return result.toString();
+    }
```
Comment (nihalzp) [inline src/main/java/athleticli/dietgoal/DietGoalList.java:34]: Good job on using StringBuilder which is good for performance

**Review by nihalzp — APPROVED** [review]
LGTM

File: src/main/java/athleticli/dietgoal/DietGoal.java (hunk 2 above)
Comment (skylee03) [reply src/main/java/athleticli/dietgoal/DietGoal.java:42]: I think this is a good suggestion: it won't hurt to re-assign `isGoalAchieved` to its current value, and it in fact can simplify the code logic.

I think changing the code to the following will simplify the code, but I'm not sure whether doing so will reduce readability:

```suggestion
        setIsGoalAchieved(currentValue >= targetValue);
```

**Pull request merged**